static/app/views/performance/newTraceDetails/traceLinksNavigation/useFindNextTrace.ts

```diff
@@ -0,0 +1,112 @@
+import {useMemo} from 'react';
+
+import type {TraceContextType} from 'sentry/components/events/interfaces/spans/types';
+import type {EventTransaction} from 'sentry/types/event';
+import {useApiQueries} from 'sentry/utils/queryClient';
+import {MutableSearch} from 'sentry/utils/tokenizeSearch';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useSyncedLocalStorageState} from 'sentry/utils/useSyncedLocalStorageState';
+import {useSpansIndexed} from 'sentry/views/insights/common/queries/useDiscover';
+import {SpanIndexedField} from 'sentry/views/insights/types';
+import {TRACE_FORMAT_PREFERENCE_KEY} from 'sentry/views/performance/newTraceDetails/traceHeader';
+
+import type {ConnectedTraceConnection} from './traceLinkNavigationButton';
+
+/**
+ *  A temporary solution for getting "next trace" data.
+ *  As traces currently only include information about the previous trace, the next trace is obtained by
+ *  getting all root spans in a certain timespan (e.g. 1h) and searching the root span which has the
+ *  current trace as it's "previous trace".
+ */
+export function useFindNextTrace({
+  direction,
+  currentTraceID,
+  linkedTraceStartTimestamp,
+  linkedTraceEndTimestamp,
+  projectID,
+}: {
+  direction: ConnectedTraceConnection;
+  currentTraceID?: string;
+  linkedTraceEndTimestamp?: number;
+  linkedTraceStartTimestamp?: number;
+  projectID?: string;
+}): TraceContextType | undefined {
+  const {data: indexedSpans} = useSpansIndexed(
+    {
+      limit: direction === 'next' && projectID ? 100 : 1,
```

Comment:
m: A limit of 100 feels a bit low. If I think about docs or sentry, we get more than 100 transactions within an hour. I know we probably need a limit for now but would it be "reasonable" to set it to something like 1k or even 10k? Really just asking, I have no answer here 😬 😅

Reply:
True - `100` is the maximum here but I can paginate the request.

Reply:
oh I see.. I mean we can also just go with 100 and see if we get any results from dogfooding. I'm not sure how this data is ordered. If the spans are ordered by time and we assume that subsequent navigations happen quickly, we should get some hits within 100. Probably good enough for demo purposes 😅